### DISCUSSION TITLE
Will Slugify be migrated to v5?

### DISCUSSION THREAD

**Discussion by yarapolana** [adonisjs/core#1654] [Help]
Hey there,
Does anyone know how to set up slug logic in v5? in v4 I used the official @Adonis/Lucid/Slugify but I haven't seen its light in v5. Is there a way to do it without?

**Comment by thetutlage** [comment]
Yes it will be. Infact, it will be a quite a simple task to do. Lemme see, if I can pick it up as part of the upcoming release

**Reply by yarapolana** [reply]
Thank you for the quick reply. =) looking forward!!

**Comment by mxthevs** [comment]
Here is what I'm doing to generate unique slugs while `adonis-lucid-slugify` doesn't get migrated to V5. It's a very simple version of the package, but it works.

```ts
import { LucidModel } from '@ioc:Adonis/Lucid/Model'
import slugify from '@slynova/slug'

export default class Slug {
  public static async make<T extends LucidModel>(
    modelInstance: T,
    fieldValue: string
  ): Promise<string> {
    const slug: string = slugify(fieldValue)

    const alreadyExists = await modelInstance
      .query()
      .select('*')
      .whereRaw(`?? ~* ?`, ['slug', `^${slug}(-[0-9]*)?$`]) // PostgreSQL only
      .orderBy('created_at', 'desc')
      .first()

    if (!alreadyExists) {
      return slug
    }

    const lastNum = Number(alreadyExists['slug'].replace(`${slug}-`, ''))
    return !lastNum || Number.isNaN(lastNum) ? `${slug}-1` : `${slug}-${lastNum + 1}`
  }
}
```
```ts
@beforeSave()
  public static async generateSlug(customer: Customer) {
    if (customer.$dirty.name) {
      customer.slug = await Slug.make(this, customer.name)
    }
  }
```

**Reply by yarapolana** [reply]
@mxthevs Thank you for sharing this =)
@thetutlage Any new on the official one?

**Reply by thetutlage** [reply]
Yeah, didn't find much time to get the out. Not sure on the timeline, but will try to prioritize it
